src/Microsoft.Azure.Batch.SoftwareEntitlement.Server/Controllers/SoftwareEntitlementsController.cs

```diff
@@ -50,15 +50,41 @@ public SoftwareEntitlementsController(Options options, ILogger logger)
         [HttpPost]
         [Produces("application/json")]
         public IActionResult RequestEntitlement(
-            [FromBody] SoftwareEntitlementRequest entitlementRequest)
+            [FromBody] SoftwareEntitlementRequest entitlementRequest,
+            [FromQuery(Name = "api-version")] string apiVersion)
         {
-            _logger.LogInformation("Request entitlement for {application}", entitlementRequest.ApplicationId);
+            if (!IsValidApiVersion(apiVersion))
+            {
+                _logger.LogError(
+                    "Selected api-version of {apiversion} is not supported; denying entitlement request.",
```

Comment:
As discussed offline, have modified placeholders to match the official SeriLog standard of PascalCase.